lib/cloudera-manager/service.rb

```diff
@@ -1,4 +1,64 @@
 module ClouderaManager
   class Service < BaseResource
+    class RestartFailedException < BaseException; end
+
+    belongs_to :cluster
+    has_many :roles
+    primary_key :name
+    collection_path "clusters/:cluster_id/services"
+    after_find { |s| s.cluster_id = s.clusterRef[:clusterName] }
+
+    BULK_RESTART_SAFE_ROLE_TYPES = [
+      "IMPALAD",
+      "DATANODE",
+      "NODEMANAGER"
+    ]
+
+    RESTART_ORDER = {
+      "YARN" => ["JOBHISTORY", "NODEMANAGER", "RESOURCEMANAGER"],
+      "HDFS" => ["SECONDARYNAMENODE", "HTTPFS", "FAILOVERCONTROLLER", "NFSGATEWAY", "DATANODE", "NAMENODE", "JOURNALNODE"],
```

Comment:
I agree, I also think it's better to fail over to then new config, so standby first